R/create_output_type_item.R

```diff
@@ -423,17 +483,29 @@ create_output_type_dist <- function(
     )
   }
 
-  check_prop_not_all_null(required, optional)
-  check_prop_type_const(required, optional)
-  check_prop_dups(required, optional)
+  if (pre_v4) {
+    check_prop_not_all_null(required, optional)
+    check_prop_type_const(required, optional)
+    check_prop_dups(required, optional)
 
-  output_type_id <- list(output_type_id = list(
-    required = required,
-    optional = optional
-  ))
+    output_type_id <- list(
+      output_type_id = list(
+        required = required,
+        optional = optional
+      )
+    )
+  } else {
```

Comment:
To address @elray1's comment above. 

```suggestion
  } else {
    if (is.null(required)) {
      cli::cli_abort(c("x" = "The argument {.arg required} cannot be NULL."))
    }
```

Reply:
This is already checked by the `check_properties_*()` fns which check against the schema so not necessary. See added test in 5da5ddd9e4790f0120d57a3761b37a042db37f23